### DISCUSSION TITLE
opengl Environment setup, Error when compiling glad undefined reference to `xxx'

### DISCUSSION THREAD

**Discussion by blankxiao** [microsoft/vcpkg#40910] [Q&A]
我在尝试学习opengl 苦于cpp项目的环境配置 希望使用vcpkg快速搭建环境 但下载glad时编译出现问题

[CmakeLists.txt](https://github.com/user-attachments/files/16963576/CmakeLists.txt)
[CMakePresets.json](https://github.com/user-attachments/files/16963579/CMakePresets.json)
[CMakeUserPresets.json](https://github.com/user-attachments/files/16963581/CMakeUserPresets.json)
[vcpkg.json](https://github.com/user-attachments/files/16963582/vcpkg.json)
[vcpkg-configuration.json](https://github.com/user-attachments/files/16963583/vcpkg-configuration.json)

**Comment by blankxiao** [comment]
I am trying to learn opengl and I am struggling with the environment configuration of the cpp project. I hope to use vcpkg to quickly build the environment, but there is a compilation problem when downloading glad.

Thanks for all the help


[build] D:/softwares/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/softwares/vcpkg/vcpkg/installed/x64-windows/debug/lib/glad.lib(CMakeFiles/glad.dir/src/glad.c.obj):(.rtc$IMZ+0x0): undefined reference to `_RTC_InitBase'
[build] D:/softwares/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/softwares/vcpkg/vcpkg/installed/x64-windows/debug/lib/glad.lib(CMakeFiles/glad.dir/src/glad.c.obj):(.rtc$TMZ+0x0): undefined reference to `_RTC_Shutdown'

**Comment by Osyotr** [comment]
Looks like you're using mingw but vcpkg buildt libraries for msvc. Change your triplet from `x64-windows` to `x64-mingw-dynamic`.

**Reply by blankxiao** [reply]
thanks for your help. I try to update my VCPKG_DEFAULT_TRIPLET environment variable from x64-windows to x64-mingw-dynamic, but 
vcpkg could not install the dependencies

```shell
PS C:\Users\lenovo> $env:VCPKG_DEFAULT_TRIPLET
x64-mingw-dynamic

PS D:\Codes\openGL\vcpkg_test> vcpkg install
Detecting compiler hash for triplet x64-windows...
-- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:7890".
Compiler found: D:/softwares/vs/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe
Detecting compiler hash for triplet x64-mingw-dynamic...
-- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:7890".
Compiler found: D:/softwares/mingw64/bin/x86_64-w64-mingw32-g++.exe
The following packages will be built and installed:
    glfw3:x64-mingw-dynamic@3.4#1 -- C:\Users\lenovo\AppData\Local\vcpkg\registries\git-trees\d4bbd4f27c2ca619f57778985120cddcedadb1cc
Restored 0 package(s) from C:\Users\lenovo\AppData\Local\vcpkg\archives in 130 us. Use --debug to see more details.
Installing 1/1 glfw3:x64-mingw-dynamic@3.4#1...
Building glfw3:x64-mingw-dynamic@3.4#1...
D:\softwares\vcpkg\vcpkg\triplets\community\x64-mingw-dynamic.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
C:\Users\lenovo\AppData\Local\vcpkg\registries\git-trees\d4bbd4f27c2ca619f57778985120cddcedadb1cc: info: installing overlay port from here       
-- Using cached glfw-glfw-3.4.tar.gz.
-- Cleaning sources at D:/softwares/vcpkg/vcpkg/buildtrees/glfw3/src/3.4-2448ff4533.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/softwares/vcpkg/vcpkg/downloads/glfw-glfw-3.4.tar.gz
-- Using source at D:/softwares/vcpkg/vcpkg/buildtrees/glfw3/src/3.4-2448ff4533.clean
-- Found external ninja('1.12.1').
-- Configuring x64-mingw-dynamic
-- Building x64-mingw-dynamic-dbg
-- Building x64-mingw-dynamic-rel
-- Fixing pkgconfig file: D:/softwares/vcpkg/vcpkg/packages/glfw3_x64-mingw-dynamic/lib/pkgconfig/glfw3.pc
-- Using cached msys2-mingw-w64-x86_64-pkgconf-1~2.2.0-1-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.5.3-3-x86_64.pkg.tar.zst.
-- Using msys root at D:/softwares/vcpkg/vcpkg/downloads/tools/msys2/1e74ca60daa10104
error: building glfw3:x64-mingw-dynamic failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle glfw3:x64-mingw-dynamic: 15 s
```
i try to remove glfw3 in vcpkg.json, the command runn successed, but obviously, cmake still could not work.
Should I specify the version when downloading here? How to specify the version?
thanks again
